plugin/trino-hive-hadoop2/src/test/java/io/trino/plugin/hive/s3select/TestHiveFileSystemS3SelectPushdownWithSplits.java

```diff
@@ -0,0 +1,101 @@
+import org.apache.hadoop.fs.Path;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+import static io.trino.plugin.hive.s3select.S3SelectPushdownConstants.CSV_SCAN_RANGE_FILES;
+import static io.trino.plugin.hive.s3select.S3SelectPushdownConstants.CSV_SCAN_RANGE_TYPES;
+import static io.trino.plugin.hive.s3select.S3SelectPushdownConstants.PIPE_SPLIT_DELIMITER;
+import static io.trino.testing.QueryAssertions.assertEqualsIgnoreOrder;
+
+public class TestHiveFileSystemS3SelectPushdownWithSplits
+        extends AbstractTestHiveFileSystemS3
+{
+    private String host;
+    private int port;
+    private SchemaTableName tableWith3MBSplits;
+    private int maxSplitSizeMB = 1;
+    private int maxInitialSplitSizeMB = 1;
+
+    @Parameters({
+            "hive.hadoop2.metastoreHost",
+            "hive.hadoop2.metastorePort",
+            "hive.hadoop2.databaseName",
+            "hive.hadoop2.s3.awsAccessKey",
+            "hive.hadoop2.s3.awsSecretKey",
+            "hive.hadoop2.s3.writableBucket",
+            "hive.hadoop2.s3.testDirectory",
+    })
+    @BeforeClass
+    public void setup(String host, int port, String databaseName, String awsAccessKey, String awsSecretKey, String writableBucket, String testDirectory)
+    {
+        this.host = host;
+        this.port = port;
+        this.database = databaseName;
+        this.awsAccessKey = awsAccessKey;
+        this.awsSecretKey = awsSecretKey;
+        this.writableBucket = writableBucket;
+        this.testDirectory = testDirectory;
+        this.tableWith3MBSplits = new SchemaTableName(database, "trino_s3select_test_csv_scan_range_pushdown");
+
+        // Initial setup is needed for running the inherited tests
+        super.setup(host, port, database, awsAccessKey, awsSecretKey, writableBucket, testDirectory, true);
+    }
+
+    @DataProvider(name = "testSplitSize")
+    public static Object[][] splitSizeParametersProvider()
+    {
+        return new Object[][] {{3, 2}, {50, 30}};
+    }
+
+    @Override
+    protected void initHiveConfig(boolean s3SelectPushdownEnabled)
+    {
+        config = new HiveConfig()
+                .setS3SelectPushdownEnabled(s3SelectPushdownEnabled)
+                .setMaxSplitSize(DataSize.of(maxSplitSizeMB, MEGABYTE))
+                .setMaxInitialSplitSize(DataSize.of(maxInitialSplitSizeMB, MEGABYTE));
+    }
+
+    @Test(dataProvider = "testSplitSize")
+    public void testQueryPushdownWithSplitSize(int maxSplitSizeMB, int maxInitialSplitSizeMB)
+            throws Exception
+    {
+        this.maxSplitSizeMB = maxSplitSizeMB;
+        this.maxInitialSplitSizeMB = maxInitialSplitSizeMB;
+
+        // Setup update is required to use the new parametrized values
+        super.setup(host, port, database, awsAccessKey, awsSecretKey, writableBucket, testDirectory, true);
```

Comment:
Can we refactor the setup method into 2 where you have a separate method for just the split initialization that this test can use? I'm wondering if it is worth re-initializing all those objects in the original setup() method. This also feels hard to maintain if we want to add multiple tests to this class later on.

Reply:
The config object is passed on during setup to other objects creation as well. The @BeforeClass initialization is needed for the inherited tests.
Maybe I am missing something, but I don't think it should be in the scope of this PR to do such a refactoring. We already have another on-going PR with tests refactoring from you. Furthermore, these tests will need to be updated anyway with filtering queries, to stay relevant when merging @fuatbasik's PR. 
Happy to refactor this in a follow-up PR.

Reply:
@preethiratnam if you are and rest of the community is happy, lets take the test refactoring to another PR. We have 3 PRs attempting to change these tests right now, and we can maybe address refactoring once we merge them?